src/components/container/DailyDataChart/DailyDataChart.tsx

```diff
@@ -0,0 +1,270 @@
+        checkAvailability();
+        MyDataHelps.on("applicationDidBecomeVisible", checkAvailability);
+        MyDataHelps.on("externalAccountSyncComplete", checkAvailability);
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", checkAvailability);
+            MyDataHelps.off("externalAccountSyncComplete", checkAvailability);
+        }
+    }, [props.dailyDataType]);
+
+    useEffect(() => {
+        loadCurrentInterval();
+        MyDataHelps.on("applicationDidBecomeVisible", loadCurrentInterval);
+        MyDataHelps.on("externalAccountSyncComplete", loadCurrentInterval);
+
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", loadCurrentInterval);
+            MyDataHelps.off("externalAccountSyncComplete", loadCurrentInterval);
+        }
+    }, [props.intervalType, props.weekStartsOn, dateRangeContext]);
+
+    var currentDate = intervalStart;
+    var data: any[] = [];
+    var chartHasData: boolean = false;
+    if (currentData) {
+        while (currentDate < intervalEnd) {
+            var dataDay: any = {
+                day: currentDate.getDate()
+            };
+            data.push(dataDay);
+            var dayKey = getDayKey(currentDate);
+            if (currentData[dayKey] != undefined && currentData[dayKey] != null) {
+                dataDay.value = currentData[dayKey];
+                dataDay.rawValue = dataDay.value;
+                dataDay.date = currentDate;
+                if (props.valueConverter) {
+                    dataDay.value = props.valueConverter(dataDay.value);
+                }
+                chartHasData = true;
+            }
+            currentDate = add(currentDate, { days: 1 });
+        }
+    }
+
+    const GraphToolTip = ({ active, payload, label }: any) => {
+        if (active && payload && payload.length) {
+            var date = payload[0].payload.date;
+            return (
+                <div className="mdhui-daily-data-tooltip">
+                    <div className="mdhui-daily-data-tooltip-value">
+                        {props.valueFormatter ? props.valueFormatter(payload[0].payload.rawValue) : payload[0].payload.value}
+                    </div>
+                    <div className="mdhui-daily-data-tooltip-date">{format(date, 'MM/dd/yyyy')}</div>
+                </div>
+            );
+        }
+        return null;
+    };
+
+    const DayTick = ({ x, y, stroke, payload }: any) => {
+        var value = payload.value;
+        if (intervalType == "Month") {
+            let currentDate = new Date(intervalStart.getFullYear(), intervalStart.getMonth(), value);
+            return <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 15} textAnchor="middle" fontSize="12">{value}</text>;
+        } else {
+            let currentDate = intervalStart;
+            let dayOfWeek = format(currentDate, "E").substr(0, 1);
+            let month = currentDate.getMonth();
```

Comment:
I don't think `month` gets used anywhere.

Reply:
Huh...no idea what that was for.